project/yuanhenglee/README.rst

```diff
@@ -0,0 +1,179 @@
+
+### Input
+
+The system takes an audio file that needs to be transcribed. as input.
+The file can be in a standard format, such as WAV or MP3. 
+
+### Harmonic stacking using constant-Q transform. 
+
+The audio data undergoes preprocessing, beginning with applying the Constant-Q
+Transform (CQT). This transforms the audio from the time domain to the frequency
+domain, capturing the harmonic content crucial for polyphonic transcription.
+Harmonic stacking is then applied to create a comprehensive frequency
+representation for each time frame. This is a critical step for distinguishing
+individual notes in harmonically rich audio. 
+
+### Inference pre-trained CNN for note generation.
+
+The preprocessed data is fed into a Convolutional Neural Network
+(CNN) architecture. This trained model analyzes the frequency representations of
+the audio frames to generate note information. The CNN is capable of accurately
+identifying pitch information for each frame, allowing it to transcribe
+polyphonic audio.
+
+### Post-processing & Alignment
+
+Post-processing steps are performed to refine the detected notes. This involves
+tasks like note duration estimation, handling overlapping frequencies, and
+convert vector-like transcription into MIDI. Note alignment ensures that the
+generated notes are correctly timed, accurately representing the musical
+content.
+
+### Output
+
+The system produces a MIDI file as output containing the transcribed musical
+notes. 
+
+## API Description
+
+BasiCPP Pitch provides a user-friendly API for both C++ and Python, allowing
+developers to integrate the AMT capabilities into their applications.
+
+### C++ Example
+
+```cpp
+#include "BasiCPP_Pitch.h" // Include the BasiCPP Pitch header file
+
+int main() {
+    // Initialize BasiCPP Pitch
+    BasiCPP_Pitch amt;
+
+    // Specify the input audio file path
+    std::string audioFilePath = "path/to/audio/file.wav";
+
+    // Specify the output MIDI file path
+    std::string midiFilePath = "output/file.mid";
+
+    // Transcribe the audio and generate the MIDI file
+    amt.transcribeAudio(audioFilePath, midiFilePath);
+
+    return 0;
+}
+```
+
+### Python Example
+
+```python
+import basicpp_pitch # Import the BasiCPP Pitch Python module
+
+def main():
+    # Initialize BasiCPP Pitch
+    amt = basicpp_pitch.BasiCPP_Pitch()
+
+    # Specify the input audio file path
+    audio_file_path = "path/to/audio/file.wav"
+
+    # Specify the output MIDI file path
+    midi_file_path = "output/file.mid"
+
+    # Transcribe the audio and generate the MIDI file
+    amt.transcribe_audio(audio_file_path, midi_file_path)
+
+if __name__ == "__main__":
+    main()
+```
+
+## Engineering Infrastructure
+
+1. Automatic build system: `CMake`
+2. Version control: `Git`
+3. Testing framework: `Pytest`
+4. Documentation: `README`, `PyDoc`
+
+## Schedule
+
+* Week 1 (10/30):
+    - [ ] Set up the environment and project structure
+    - [ ] Test the pure Python prototype
```

Comment:
Do you have the Python prototype ready now?

Reply:
The Python prototype will be [the one](https://github.com/spotify/basic-pitch/tree/main) provided by Spotify. I think the process will be pretty much the same except that basic-pitch uses TensorFlow and Librosa, which I'm going to rewrite in C++.